File: 15-key-vault/python-sdk/key_analytics.py
"""
Cost Katana Key Vault: Analytics (Python)

Run: python 19-key-vault/python-sdk/key_analytics.py
"""

import os
import requests

API_BASE = "https://cost-katana-backend.store/api"
API_KEY = os.getenv("COST_KATANA_API_KEY")

def get_dashboard():
    response = requests.get(
        f"{API_BASE}/key-vault/dashboard",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    data = response.json()["data"]
    print("📊 Key Vault Dashboard:")
    print(f"  Provider Keys: {data['providerKeys']}")
    print(f"  Proxy Keys: {data['proxyKeys']}")
    return data

def get_analytics():
    response = requests.get(
        f"{API_BASE}/key-vault/analytics",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    data = response.json()["data"]
    print("📈 Analytics:")
    print(f"  Total Requests: {data['totalRequests']}")
    print(f"  Total Cost: ${data['totalCost']}")
    return data

if __name__ == "__main__":
    print("🥷 Key Vault: Analytics\n")
    get_dashboard()
    get_analytics()

File: 15-key-vault/npm-package/key-analytics.ts
/**
 * Cost Katana Key Vault: Analytics
 * 
 * Run: npx ts-node 19-key-vault/npm-package/key-analytics.ts
 */

import axios from 'axios';

const API_BASE = 'https://api.costkatana.com/api'\;
const API_KEY = process.env.COST_KATANA_API_KEY;

async function getKeyVaultDashboard() {
  const response = await axios.get(
    `${API_BASE}/key-vault/dashboard`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log('📊 Key Vault Dashboard:');
  console.log(`  Provider Keys: ${response.data.data.providerKeys}`);
  console.log(`  Proxy Keys: ${response.data.data.proxyKeys}`);
  return response.data.data;
}

async function getProxyKeyAnalytics() {
  const response = await axios.get(
    `${API_BASE}/key-vault/analytics`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log('📈 Proxy Key Analytics:');
  console.log(`  Total Requests: ${response.data.data.totalRequests}`);
  console.log(`  Total Cost: $${response.data.data.totalCost}`);
  return response.data.data;
}

async function main() {
  console.log('🥷 Key Vault: Analytics\n');
  await getKeyVaultDashboard();
  await getProxyKeyAnalytics();
}

if (require.main === module) main();
export { getKeyVaultDashboard, getProxyKeyAnalytics };
